plugins/com.google.cloud.tools.eclipse.appengine.validation/src/com/google/cloud/tools/eclipse/appengine/validation/AppEngineWebXmlValidator.java

```diff
@@ -16,96 +16,40 @@
 
 package com.google.cloud.tools.eclipse.appengine.validation;
 
-import com.google.common.io.ByteStreams;
 import java.io.IOException;
-import java.io.InputStream;
-import java.util.logging.Level;
-import java.util.logging.Logger;
 import java.util.Map;
 import javax.xml.parsers.ParserConfigurationException;
-import org.eclipse.core.resources.IFile;
+
 import org.eclipse.core.resources.IMarker;
 import org.eclipse.core.resources.IResource;
 import org.eclipse.core.runtime.CoreException;
-import org.eclipse.core.runtime.IProgressMonitor;
-import org.eclipse.wst.validation.AbstractValidator;
-import org.eclipse.wst.validation.ValidationEvent;
-import org.eclipse.wst.validation.ValidationResult;
-import org.eclipse.wst.validation.ValidationState;
 import org.xml.sax.SAXException;
-import org.xml.sax.SAXParseException;
 
 /**
  * Validator for appengine-web.xml
  */
-public class AppEngineWebXmlValidator extends AbstractValidator {
-  
-  private static final Logger logger = Logger.getLogger(
-      AppEngineWebXmlValidator.class.getName());
-  
-  /**
-   * Extracts byte[] from appengine-web.xml. 
-   */
-  @Override
-  public ValidationResult validate(ValidationEvent event, ValidationState state,
-      IProgressMonitor monitor) {
-    IResource resource = event.getResource();
-    IFile file = (IFile) resource;
-    try (InputStream in = file.getContents()) {
-      byte[] bytes = ByteStreams.toByteArray(in);
-      validate(resource, bytes);
-      return new ValidationResult();
-    } catch (IOException | CoreException | ParserConfigurationException ex) {
-      logger.log(Level.SEVERE, ex.getMessage());
-      return new ValidationResult();
-    }
-  }
+public class AppEngineWebXmlValidator extends AbstractXmlValidator {
     
   /**
-   * Clears all problem markers from the resource, then adds a marker to 
+   * Clears all problem markers from the resource, then adds an application marker to 
    * appengine-web.xml for every {@link BannedElement} found in the file.
    */
-  static void validate(IResource resource, byte[] bytes) 
+  @Override
+  protected void validate(IResource resource, byte[] bytes) 
       throws CoreException, IOException, ParserConfigurationException {
     try {
-      deleteApplicationMarkers(resource);
+      deleteMarkers(resource);
```

Comment:
I'll just note that this causes churn as markers are deleted and created.  It's better (but more work) to determine new marker locations and then merge the results with the existing markers. But this is fine for now.